code-generation/llms-vscode-integration.md

```diff
@@ -0,0 +1,85 @@
+
+Once the Model Service is operational, configure the "Continue" extension. Open VS Code and access the Command Palette by pressing `Ctrl+Shift+P` (Windows/Linux) or `Cmd+Shift+P` (Mac). Type "Continue: Configure" and select the option to open the configuration file. Edit the `config.json` file with the appropriate configuration.
+
+```
+{
+  "title": "YourTitleHere",
+  "model": "YourModelName",
+  "completionOptions": {},
+  "apiBase": "http://localhost:8001/v1/",
+  "provider": "openai"
+}
+```
+
+## Step 4: Examples of Interaction
+
+Now that you've configured the "Continue" extension, let's explore how you can effectively interact with the language model directly within VS Code. Here are several ways to engage with the extension:
+
+1. **Prompting for code generation:** Open the "Continue" panel in VS Code and prompt the extension with a specific task, such as "Write a code to add two numbers." The extension will then provide relevant code autocompletion suggestions based on your input prompt, aiding in code generation and text completion tasks.
+
+   ![Prompt-response](../assets/interaction-vscode1.png)
+
+2. **Querying Working Code:** Copy your existing code snippet or press `⌘ + L` to paste it into the "Continue" panel, then pose a question such as "Explain this section of the code." The extension (LLM) will analyze the code snippet and provide explanations or insights to help you understand it better.
+
+   ![Querying Working Code](../assets/interaction-vscode2.png)
+
+3. **Editing Code in Script:** Edit your Python code directly within a `.py` script file using the "Continue" extension. Press `⌘ + I` to initiate the edit mode. You can then refine a specific line of code or request enhancements to make it more efficient. The extension will suggest additional code by replacing your edited code and provide options for you to accept or reject the proposed changes.
+
+   ![Editing Code in Script](../assets/interaction-vscode3.png)
+
+By exploring these interactions, users can fully leverage the capabilities of language models within VS Code, enhancing their coding experience and productivity.
+
+4. **Tab Autocomplete:**
+
+![autocompletion-config-example](../assets/autocompletion-example.png)
+
+ The "Continue" extension also offers a tab autocomplete feature in its pre-release version. To set up autocomplete with a custom model, you'll need to edit the `config.json` file as shown below:
+
+```
+{
+    "tabAutocompleteModel": {
+        "title": "Tab Autocomplete Model",
+        "provider": "provider name",
+        "model": "model name",
+        "apiBase": "https://<endpoint>"
+    },
+    ...
+}
+```
+
+The tabAutocompleteModel is similar to other objects in the models array of config.json. You have the flexibility to choose any model you prefer, but it's recommended to use a small model for tab autocomplete, such as deepseek-1b, starcoder-1b,  starcoder-3b, or stable-code-3b for optimal performance.
```

Comment:
Do any of these other models have MIT or Apache-2.0 License?

Reply:
Deepseek has the MIT:
- https://huggingface.co/deepseek-ai/deepseek-vl-1.3b-chat
- https://github.com/deepseek-ai/DeepSeek-LLM/blob/HEAD/LICENSE-CODE

starcoder has the BigCode OpenRAIL-M v1 license:
- https://huggingface.co/bigcode/starcoder2-15b#license
- explained here: https://www.bigcode-project.org/docs/pages/model-license/
`Open Responsible AI Licenses (OpenRAIL) are licenses designed to permit free and open access, re-use, and downstream distribution of derivatives of AI artifacts, for research, commercial or non-commercial purposes, as long as the use restrictions present in the license always apply (including to derivative works).`
- use restrictions: https://huggingface.co/spaces/bigcode/bigcode-model-license-agreement#attachment-a-use-restrictions

And stable-code has a non-commercial research community license
- https://huggingface.co/stabilityai/stable-code-3b#model-details

Reply:
Welp I didn't completely read the Deepseek licensing, the code in their [repository](https://github.com/deepseek-ai/DeepSeek-LLM/tree/6712a86bfb7dd25c73383c5ad2eb7a8db540258b) is under the MIT license but 

"The use of DeepSeek-VL Base/Chat models is subject to [DeepSeek Model License](https://github.com/deepseek-ai/DeepSeek-LLM/blob/HEAD/LICENSE-MODEL). DeepSeek-VL series (including Base and Chat) supports commercial use."

Reply:
No problem! I'll also note that our colleagues from Germany have done some work around integrating Ollama with Continue in disconnected environments utilizing OpenShift dev spaces if it helps the awesome work y'all are doing: https://www.opensourcerers.org/2023/11/06/a-personal-ai-assistant-for-developers-that-doesnt-phone-home/